packages/react/src/components/loaded/loaded.tsx

```diff
@@ -0,0 +1,32 @@
+import { SDKState } from "../../context/slash-id-context";
+import { useSlashID } from "../../main";
+
+interface Props {
+    fallback?: React.ReactNode
+    children: React.ReactNode
+}
+
+/**
+ * Conditional rendering helper
+ * 
+ * Acts as a guard for SlashID core SDK dependent operations
+ */
+export const SlashIDLoaded: React.FC<Props>  = ({ fallback, children }) => {
+    const { sdkState } = useSlashID()
+
+    if (sdkState !== SDKState.Ready) {
+        if (!fallback) return null
+
+        return (
+            <>
+                {fallback}
+            </>
+        )
```

Comment:
Just curious, why do you need the <>?

Wouldn't `return fallback` work?

Reply:
I definitely agree that it's verbose and pretty ugly.

Technically speaking yes we can `return fallback` because we do a falsy check on line 18, so `fallback` must be a truthy `React.ReactNode` - so we can return it, surely?

In practice this doesn't work because `React.ReactNode` is not perfectly compatible with the `React.FC<...>` expected return type.

```
Type '({ fallback, children }: Props) => string | number | true | ReactFragment | Element | null' is not assignable to type 'FC<Props>'.
  Type 'string | number | true | ReactFragment | Element | null' is not assignable to type 'ReactElement<any, any> | null'.
    Type 'string' is not assignable to type 'ReactElement<any, any>'.
```

By wrapping `fallback` [in a fragment](https://react.dev/reference/react/Fragment) we comply with the return type and cover a bunch of edge cases.

Reply:
@jake-slashid React.FC is unnecessary (and I think it's still discouraged by the community?) for the most part, as TS can infer the type quite well. I know that we're using it quite heavily, but maybe it's time to stop doing it for the new components? cc @ikovic

Reply:
Full transparency: I don't have a strong opinion about this, I seen it used in another component and was following that.

Reply:
By now I think it's rather clear that `React.FC` is mostly abandoned when compared to the following:
- adding an explicit children field in the props type declaration in case the component really needs children
- adding an explicit return type like JSX.Element

My vote would be to move to that approach as we work on maintaining the existing components and implementing the new ones.

Reply:
Thanks for your input! Here's the PR where it gets soft-deprecated: https://github.com/facebook/create-react-app/pull/8177

I'll remove React.FC and simplify the component. 🙌

Reply:
done